app/modules/template/service.server.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  AssetStatus,
+  type Organization,
+  type Template,
+  type User,
+} from "@prisma/client";
+import { db } from "~/database";
+import { getPublicFileURL, parseFileFormData } from "~/utils/storage.server";
+import { createNote } from "../asset";
+
+export async function createTemplate({
+  name,
+  type,
+  description,
+  signatureRequired,
+  userId,
+  organizationId,
+}: Pick<Template, "name" | "type" | "description" | "signatureRequired"> & {
+  userId: User["id"];
+  organizationId: Organization["id"];
+}) {
+  // Count the number of templates of same type for the user
+  const sameExistingTemplateCount = await db.template.count({
+    where: { type, userId },
+  });
+
+  const data = {
+    name,
+    type,
+    description,
+    signatureRequired,
+    creator: {
+      connect: {
+        id: userId,
+      },
+    },
+    organization: {
+      connect: {
+        id: organizationId,
+      },
+    },
+    isDefault: sameExistingTemplateCount === 0,
+  };
+
+  const template = await db.template.create({ data });
+
+  return template;
+}
+
+export async function updateTemplate({
+  id,
+  name,
+  description,
+  signatureRequired,
+  userId,
+}: Pick<Template, "id" | "name" | "description" | "signatureRequired"> & {
+  userId: User["id"];
+}) {
+  const data = {
+    name,
+    description,
+    signatureRequired,
+  };
+
+  const updatedTemplate = await db.template.update({
+    where: { id },
+    data,
+  });
+
+  /**
+   * If the signatureRequired is true, we need to search through all the Custodies that
+   * have this tempalate associated with it. We will check if the templateSigned is false.
+   *
+   * If it is false, this could mean a scenario that the custodian has the asset in custody
+   * and wasn't required to sign the template. But since we are setting signatureRequired to true,
+   * we need to set the asset custory to "AVAILABLE" and furthermore, ask the custodian to sign
+   * the template via mailing them.
+   */
+  if (signatureRequired === true) {
+    const custodies = await db.custody.findMany({
```

Comment:
This is a very interesting case. But I am not sure this is the best way to handle this. 
@jurrejansen @carlosvirreira we need to take some decisions here. Here is the case:
1. Template exists and doesnt require signature and some people already have custodies assigned through that template.
2. The template gets update to signature required

What do we do? I think we discussed something that we just dont allow templates to be updated after they have 1 custody using them. What do we think?

Reply:
@carlosvirreira @jurrejansen what do you guys think about this? We discussed that if a template is used even once, we should not allow for the file to be changed, but should we consider this also for the case of changing required? I would say yes.